### PR TITLE
Add Velodrome

### PR DESCRIPTION
**Pull request by peachbits**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204241528085405

### CODE REVIEW

**Review by swansontec — APPROVED** [review]
It looks like this will all work, but I see a few small things that could be tidied.

File: src/swap/defi/uni-v2-based/uniV2Utils.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export const getSwapAmounts = async (
   router: Contract,
   quoteFor: string,
   nativeAmount: string,
-  path: string[],
+  path: any,
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/swap/defi/uni-v2-based/uniV2Utils.ts]: fixed

File: src/swap/defi/uni-v2-based/uniV2Utils.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ export const getSwapAmounts = async (
 export const getSwapTransactions = async (
   provider: ethers.providers.Provider,
   inOutAddresses: InOutTokenAddresses,
-  path: string[],
+  path: any,
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/swap/defi/uni-v2-based/uniV2Utils.ts]: fixed

File: src/swap/defi/uni-v2-based/plugins/velodrome.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+      request.quoteFor === 'to' ||
+      // Velodrome only supports Optimism
+      fromWallet.currencyInfo.pluginId !== 'optimism' ||
+      toWallet.currencyInfo.pluginId !== 'optimism'
+    ) {
+      throw new SwapCurrencyError(
+        swapInfo,
+        request.fromCurrencyCode,
+        request.toCurrencyCode
+      )
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/swap/defi/uni-v2-based/plugins/velodrome.ts]: fixed

File: src/swap/defi/uni-v2-based/plugins/velodrome.ts [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    }
+
+    // Sanity check: Both wallets should be of the same chain.
+    if (
+      fromWallet.currencyInfo.currencyCode !==
+      toWallet.currencyInfo.currencyCode
+    )
+      throw new Error(`${swapInfo.displayName}: Mismatched wallet chain`)
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/swap/defi/uni-v2-based/plugins/velodrome.ts]: fixed

File: src/swap/defi/uni-v2-based/plugins/velodrome.ts [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+import VELODROME_V1_ROUTER_ABI from '../../abi/VELODROME_V1_ROUTER_ABI'
+import WRAPPED_OPTIMISM_ETH_ABI from '../../abi/WRAPPED_OPTIMISM_ETH_ABI'
+import { getInOutTokenAddresses } from '../../defiUtils'
+import { getSwapAmounts, getSwapTransactions } from '../uniV2Utils'
+
+const swapInfo: EdgeSwapInfo = {
+  pluginId: 'velodrome',
+  isDex: true,
+  displayName: 'Velodrome',
+  supportEmail: 'support@edge.app'
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/swap/defi/uni-v2-based/plugins/velodrome.ts:30]: leaving since we use our email for all dex plugins

**Pull request merged**